### DISCUSSION TITLE
How do I Deploy Remix App on Windows IIS shared hosting?

### DISCUSSION THREAD

**Discussion by apcjs** [remix-run/remix#6802] [General]
I want to deploy a Remix app on a Windows IIS Server. There have been answers to this question here - [How to deploy react Remix framework to IIS on windows server 2022?](https://stackoverflow.com/questions/75049232/)

However, this requires the server to be logged in. If I'm using a VPS Server that is possible. But how do I deploy to a shared hosting, for example, where I cannot run npm? Assumming I have IIS node installed on the shared hosting website, how do I deploy a Remix build?

A create-react-app webiste can be easily deployed by just copying the dist folder to the server.

When I build a remix app. I get a public/build folder. How do I deploy this folder to a public IIS website.

**Comment by sergiodxa** [comment]
You could install and build locally, then upload the node_modules folder, package.json, public folder, and your server files (your server entry point if you have one, the build and any other file you may need).

Once you upload everything, make your server run npm start.

**Reply by arundish** [reply]
Thanks for the answer. I do not have access to run npm start on a shared hosting server.

Also some of my customers do not want to host on a public cloud. They host all their apps internally on a Windows IIS server. I have built a number of React apps for then with create-react-app. They just copy the dist folder to a IIS folder and it works. No need to run npm start. The server can be logged off and its fine. Any changes, we just push the dist folder over FTP and its done

If this is not possible, I will surely not be able to go ahead with Remix. I like the framework though.

**Reply by kiliman** [reply]
Remix is a server framework and will require `node` to run. All your previous `create-react-app` apps generated static HTML and JavaScript that IIS serves like any other file.

So unless your shared hosting allows you to run `node` apps, you won't be able to use Remix. However, you can still use React Router and get the benefits of Remix, like loaders and actions, nested layouts, etc.

**Reply by arundish** [reply]
Thanks. Node is installed on the server. My other apps use Express and they work fine.

Can you tell me which folder I need to deploy on the server. I tried copying the files in the public/build folder. There are a couple of JS files there `entry.client*.js` and `root*.js`. Which of these files are the entry point for the web site? Unlike a `create-react-app` build, I do not find a `index.html` file

**Reply by lili21** [reply]
the project folder is what you need to deploy to the server.

**Reply by lili21** [reply]
you can use the express template, and the server.js file is the entry point.

**Reply by alpemreelmas** [reply]
> So unless your shared hosting allows you to run `node` apps, you won't be able to use Remix. However, you can still use React Router and get the benefits of Remix, like loaders and actions, nested layouts, etc.

As far as I understood, we can use remix.js using client side rendering on shared hosting. And then build it, put files on shared hosting. Is that correct ?

**Comment by MuhammadDanishSiddiqui1988** [comment]
how to remix Application easy to deploy after build. windows and plesk windows control panel. step by step deploy or how can folder deploy
